dockerplugin.py

```diff
@@ -412,11 +465,88 @@ def read_callback(self):
+                            .format(container=_c(container), msg=e))
+                # Print stack trace to log if verbose logging is on
+                log.info(traceback.format_exc())
+
+
+class CollectdLogHandler(logging.Handler):
+    """CollectdLogHandler
+    A custom log handler that forwards log messages raised
+    at level debug, info, notice, warning, and error
+    to collectd's built in logging.  Suppresses extraneous
+    info and debug statements using a "verbose" boolean
+    """
+    def __init__(self, plugin="unknown", verbose=False):
+        """@constructor
+        @param plugin : str
+        @param verbose : str
+        """
+        self.verbose = verbose
+        self.plugin = plugin
+        logging.Handler.__init__(self, level=logging.NOTSET)
+
+    def emit(self, record):
+        """
+        Emits a log record to the appropraite collectd log function
+        @param record : obj
+        """
+        try:
+            if record.levelname == 'ERROR':
+                if record.msg is not None:
+                    collectd.error('%s : %s' % (self.plugin, record.msg))
+            elif record.levelname == 'WARNING':
+                if record.msg is not None:
+                    collectd.warning('%s : %s' % (self.plugin, record.msg))
+            elif record.levelname == 'NOTICE':
+                if record.msg is not None:
+                    collectd.notice('%s : %s' % (self.plugin, record.msg))
+            elif record.levelname == 'INFO':
+                if self.verbose is True and record.msg is not None:
+                    collectd.info('%s : %s' % (self.plugin, record.msg))
+            elif record.levelname == 'DEBUG':
+                if self.verbose is True and record.msg is not None:
+                    collectd.debug('%s : %s' % (self.plugin, record.msg))
+        except Exception as e:
+            collectd.warning(('{p} [ERROR]: Failed to write log statement due '
+                              'to: {e}').format(p=self.plugin,
+                                                e=e
+                                                ))
+
 
+class CollectdLogger(logging.Logger):
+    """CollectdLogger
+    Custom python logger that forwards log statements at
+    level: debug, info, notice, warning, error
+    """
+    def __init__(self, name, level=logging.NOTSET):
+        """@constructor
+        @param name : str
+        @param level : str
+        """
+        logging.Logger.__init__(self, name, level)
+        logging.addLevelName(25, 'NOTICE')
+
+    def notice(self, msg):
+        """
+        Logs a 'NOTICE' level statement at level 25
+        @param msg : str
+        """
+        self.log(25, msg)
+
+
+# Set logger class
+logging.setLoggerClass(CollectdLogger)
+# Get logger
+log = logging.getLogger(__name__)
+# Set log level
+log.setLevel(logging.DEBUG)
+# Instantiate log handler
+handle = CollectdLogHandler('docker')
+# Add handler to logger
```

Comment:
Those comments lines 540-548 are useless as they just paraphrase what the code is doing.